sequencer/src/state.rs

```diff
@@ -1,19 +1,27 @@
-use crate::{chain_variables::ChainVariables, Error, Header, Payload};
+use crate::{
+    block::{BlockMerkleTree, FeeAccount, FeeAmount, FeeMerkleTree, _validate_proposal, ValidatedState},
+    Error, Header, Payload,
+};
 use commit::{Commitment, Committable};
-use ethers::prelude::BlockNumber;
 use hotshot::traits::State as HotShotState;
-use hotshot_types::data::ViewNumber;
-use serde::{Deserialize, Serialize};
-use std::fmt::Debug;
-
-#[derive(Default, Serialize, Deserialize, Clone, Debug, Hash, PartialEq, Eq)]
-pub struct State {
-    chain_variables: ChainVariables,
-    ethereum_block_tag: BlockNumber,
+use hotshot_types::data::{ViewNumber, BlockError};
+use jf_primitives::merkle_tree::{MerkleTreeScheme, UniversalMerkleTreeScheme};
+
+impl Default for ValidatedState {
```

Comment:
Where is `ValidatedState` defined? IMO it would make sense to define it right here before the trait ipls

Reply:
Yea I was just using the one we previously defined in `block.rs` to make minimum changes. but I will move it here.